contracts/core/modules/RebalancingSetExchangeIssuanceModule.sol

```diff
@@ -134,313 +137,452 @@ contract RebalancingSetExchangeIssuanceModule is
+        // Expect rebalancing SetToken to be valid and enabled SetToken
+        require(
+            coreInstance.validSets(_rebalancingSetAddress),
+            "RebalancingSetExchangeIssuance.validateInputs: Invalid or disabled SetToken address"
+        );
+
+        require(
+            _rebalancingSetQuantity > 0,
+            "RebalancingSetExchangeIssuance.validateInputs: Quantity must be > 0"
+        );
+        
+        // Make sure Issuance quantity is multiple of the rebalancing SetToken natural unit
+        require(
+            _rebalancingSetQuantity.mod(ISetToken(_rebalancingSetAddress).naturalUnit()) == 0,
+            "RebalancingSetExchangeIssuance.validateInputs: Quantity must be multiple of natural unit"
+        );
 
-        // Calculate the possible number of Sets issuable (may not be a multiple of natural unit)
-        uint256 possibleIssuableRBSetQuantity = _baseSetIssueQuantity.mul(rbSetNaturalUnit).div(rbSetUnitShares);
+        // Only 1 receive token in redeem and 1 send token in issue allowed
+        require(
+            _transactTokenArray.length == 1,
+            "RebalancingSetExchangeIssuance.validateInputs: Only 1 Send/Receive Token Allowed"
+        );
 
-        // Ensure that the base Set quantity is a multiple of the rebalancing Set natural unit
-        uint256 rbSetIssueQuantity = possibleIssuableRBSetQuantity.div(rbSetNaturalUnit).mul(rbSetNaturalUnit);
+        require(
+            _transactTokenAddress == _transactTokenArray[0],
+            "RebalancingSetExchangeIssuance.validateInputs: Send/Receive token must match required"
+        );
 
-        return rbSetIssueQuantity;
-    }
+        // Validate that the base Set address matches the issuanceParams Set Address
+        address baseSet = ISetToken(_rebalancingSetAddress).getComponents()[0];
+        require(
+            baseSet == _baseSetAddress,
+            "RebalancingSetExchangeIssuance.validateInputs: Base Set addresses must match"
+        );
+    } 
 
     /**
-     * Validate that the redeem parameters and inputs are congruent.
+     * Issue a Rebalancing Set using a specified ERC20 payment token. The payment token is used in ExchangeIssue
+     * to acquire the base SetToken components and issue the base SetToken. The base SetToken is then used to
+     * issue the Rebalancing SetToken. The payment token can be utilized as a component of the base SetToken.
+     * All remaining tokens / change are flushed and returned to the user.
      *
-     * @param  _rebalancingSetAddress    Address of the rebalancing Set
-     * @param  _rebalancingSetQuantity   Quantity of rebalancing Set to redeem
-     * @param  _collateralSetAddress     Address of base Set in ExchangeIssueanceParams
-     * @param  _collateralSetAddress     Address of base Set in ExchangeIssueanceParams
+     * Note: We do not validate the rebalancing SetToken quantity and the exchangeIssuanceParams base SetToken 
```

Comment:
What do you mean by this note?

Reply:
It means that we don't check that the implied base SetToken quantity from the RB Set Quantity == the passed in base SetToken quantity from exchangeIssuanceParams